agents/ibootbar/ibootbar.py

```diff
@@ -0,0 +1,520 @@
+
+        self.is_streaming = True
+
+        while self.is_streaming:
+            yield self.ibootbar.run_snmp_get(session)
+            self.log.debug("{data}", data=session.data)
+            yield dsleep(1)
+
+        return True, "Finished Recording"
+
+    def _stop_acq(self, session, params=None):
+        """_stop_acq()
+        **Task** - Stop task associated with acq process.
+        """
+        self.is_streaming = False
+        return True, "Stopping Recording"
+
+    @ocs_agent.param('_')
+    @inlineCallbacks
+    def get_names(self, session, params=None):
+        """
+        **Task** - Gets list of outlet names and updates the name list for the next get request
+        """
+        with self.lock.acquire_timeout(3, job='get_names') as acquired:
+            if acquired:
+                # Build list of the 8 outlet OIDs
+                get_list = []
+                for i in range(8):
+                    get_list.append(('IBOOTPDU-MIB', 'outletName', i))
+
+                # Clear the default names defined in self.ibootbar
+                self.ibootbar.names = []
+
+                # Send get command and append results to self.ibootbar.names
+                result = yield self.ibootbar.snmp.get(get_list, self.ibootbar.version)
+                for i in range(len(result)):
+                    self.ibootbar.names.append(result[i][1].prettyPrint())
+
+                return True, "The outlet names are: {}".\
+                    format(self.ibootbar.names)
+            else:
+                return False, "Could not acquire lock"
+
+    @ocs_agent.param('outlet', choices=[0, 1, 2, 3, 4, 5, 6, 7])
+    @ocs_agent.param('state', choices=['on', 'off'])
+    @inlineCallbacks
+    def set_outlet(self, session, params=None):
+        """
+        **Task** - Sets a particular outlet to on/off
+        """
+        with self.lock.acquire_timeout(3, job='set_outlet') as acquired:
+            if acquired:
+                # Convert given state parameter to integer
+                if params['state'] == 'on':
+                    state = 1
+                else:
+                    state = 0
+
+                # Send set command to given outlet
+                outlet = [('IBOOTPDU-MIB', 'outletControl', params['outlet'])]
+                setcmd = yield self.ibootbar.snmp.set(outlet, self.ibootbar.version, state)
+                print(setcmd)
+
+                # Force get status request by rewinding the lastGet time by 60 seconds
+                for mib in self.ibootbar.mib_timings:
+                    mib["lastGet"] = mib["lastGet"] - 60
+
+                return True, 'Set outlet {} to {}'.\
+                    format(params['outlet'], params['state'])
+            else:
+                return False, "Could not acquire lock"
+
+    @ocs_agent.param('outlet', choices=[0, 1, 2, 3, 4, 5, 6, 7])
+    @ocs_agent.param('cycle_time', default=10, type=int)
+    @inlineCallbacks
+    def cycle_outlet(self, session, params=None):
```

Comment:
One problem here is that since acq is only querying power states every minute, we likely won't see a cycle happen in the output in `session.data`.

The simplest solution would probably be to just query much more often, like once a second. Otherwise you could force more rapid updates during the cycle somehow.